### DISCUSSION TITLE
/api/tokens says internal error

### DISCUSSION THREAD

**Discussion by Luke1505** [NginxProxyManager/nginx-proxy-manager#3615] [Q&A]
after setting up my NPM i want to reach it over the API but the API Just returns a 400 - Internal error.

Has Anyone Dealt with that and can tell me how to fix this?

**Comment by Luke1505** [comment]
![image](https://github.com/NginxProxyManager/nginx-proxy-manager/assets/64462934/3dc898f6-de23-4a0a-b924-485a120fa8d8)

**Comment by gadget78** [comment]
there isnt much docs on the API... 
if you want to get a token to then use the API you need to .. 

`curl "http://192.168.178.91:81/api/tokens"   -H "Content-Type: application/json; charset=UTF-8"   --data-raw '{"identity":"email@address","secret":"PASSWORD","expiry":"2d"}'`

(or for expiry 1y, or maybe 1w ?)
